spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinder.java

```diff
@@ -108,4 +109,10 @@ protected void afterUnbindConsumer(ConsumerDestination destination, String group
 		super.afterUnbindConsumer(destination, group, consumerProperties);
 		this.pubSubChannelProvisioner.afterUnbindConsumer(destination);
 	}
+
+	@Override
+	protected PolledConsumerResources createPolledConsumerResources(String name, String group, ConsumerDestination destination, ExtendedConsumerProperties<PubSubConsumerProperties> consumerProperties) {
+		return new PolledConsumerResources(new PubSubMessageSource(this.pubSubTemplate, destination.getName()),
+				registerErrorInfrastructure(destination, group, consumerProperties, true));
```

Comment:
Well, what is the problem to go ahead with this solution?

In the end we have just this internal logic:
```
this.retryTemplate.execute(context -> {
		this.handle(message, handler);
		return null;
	}, this.recoveryCallback);
```

See `DefaultPollableMessageSource`.

I think we are good to go with the current implementation.

What am I missing?

Reply:
Yep, that looks good -- I will leave the current implementation.